packages/contracts-bedrock/src/Safe/LivenessModule.sol

```diff
@@ -0,0 +1,148 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Safe } from "safe-contracts/Safe.sol";
+import { Enum } from "safe-contracts/common/Enum.sol";
+import { OwnerManager } from "safe-contracts/base/OwnerManager.sol";
+import { LivenessGuard } from "src/Safe/LivenessGuard.sol";
+
+// TODO(maurelian): remove me
+import { console2 as console } from "forge-std/console2.sol";
+
+/// @title LivenessModule
+/// @notice This module is intended to be used in conjunction with the LivenessGuard. It should be able to
+///         execute a transaction on the Safe in only a small number of cases:
+contract LivenessModule {
+    /// @notice The Safe contract instance
+    Safe public safe;
+
+    /// @notice The LivenessGuard contract instance
+    LivenessGuard public livenessGuard;
+
+    /// @notice The interval, in seconds, during which an owner must have demonstrated liveness
+    uint256 public livenessInterval;
+
+    /// @notice The minimum number of owners before ownership of the safe is transferred to the fallback owner.
+    uint256 public minOwners;
+
+    /// @notice The fallback owner of the Safe
+    address public fallbackOwner;
+
+    // Constructor to initialize the Safe and baseModule instances
+    constructor(
+        Safe _safe,
+        LivenessGuard _livenessGuard,
+        uint256 _livenessInterval,
+        uint256 _minOwners,
+        address _fallbackOwner
+    ) {
+        safe = _safe;
+        livenessGuard = _livenessGuard;
+        livenessInterval = _livenessInterval;
+        minOwners = _minOwners;
+        fallbackOwner = _fallbackOwner;
+    }
+
+    /// @notice This function can be called by anyone to remove an owner that has not signed a transaction
+    ///         during the livness interval. If the number of owners drops below
+    function removeOwner(address owner) external {
+        // Check that the owner has not signed a transaction in the last 30 days
+        require(
+            livenessGuard.lastSigned(owner) < block.timestamp - livenessInterval,
+            "LivenessModule: owner has signed recently"
+        );
```

Comment:
Instead of storing the guard address in storage we should query the `safe` for it. This ensures the `removeOwner` method is always calling the current guard, whereas here we might be calling a guard that's no longer used, resulting in signers erroneously being removed.

Reply:
https://github.com/ethereum-optimism/optimism/blob/jm/safe-liveness/packages/contracts-bedrock/src/Safe/LivenessModule.sol#L152

Reply:
The `livenessGuard` value is now immutable. 
The `removeOwner` method will now revert if guard value in the safe doe not match.